ansible_collections/arista/avd/roles/eos_designs/python_modules/underlay/ethernet_interfaces.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.filter.list_compress import list_compress
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import default, get
+
+from .utils import UtilsMixin
+
+
+class EthernetInterfacesMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def ethernet_interfaces(self) -> dict | None:
+        """
+        Return structured config for ethernet_interfaces
+        """
+        ethernet_interfaces = {}
+        for link in self._underlay_links:
+            # common values
+            ethernet_interface = {
+                "peer": get(link, "peer"),
```

Comment:
Please use native python for where it is simple and readable. Here the "peer" key is required, so using `link["peer"]` would be better, safer and more efficient.
If "peer" is not required, then use `link.get("peer")`.
Our `get()` function should only be used for multilevel dict lookups, or if we look in the inputs, where we need to check if something required is there, and want to raise if not. (setting required=True)